src/components/icon/icon.component.tsx

```diff
@@ -1,47 +1,104 @@
 import React, { useContext } from "react";
-import PropTypes from "prop-types";
-import styledSystemPropTypes from "@styled-system/prop-types";
-import tagComponent from "../../__internal__/utils/helpers/tags/tags";
-import StyledIcon from "./icon.style";
+import { MarginProps } from "styled-system";
+import invariant from "invariant";
 import Tooltip from "../tooltip";
 import { filterStyledSystemMarginProps } from "../../style/utils";
-import { ICON_TOOLTIP_POSITIONS } from "./icon-config";
 import { TooltipContext } from "../../__internal__/tooltip-provider";
+import tagComponent from "../../__internal__/utils/helpers/tags/tags";
+import StyledIcon, { StyledIconProps } from "./icon.style";
+import { ICON_TOOLTIP_POSITIONS } from "./icon-config";
+import { IconType } from "./icon-type";
+import { TooltipPositions } from "../tooltip/tooltip.config";
 
-const marginPropTypes = filterStyledSystemMarginProps(
-  styledSystemPropTypes.space
-);
+export type LegacyIconTypes =
+  | "help"
+  | "maintenance"
+  | "new"
+  | "success"
+  | "messages";
 
-const Icon = React.forwardRef(
+export interface IconProps extends Omit<StyledIconProps, "type">, MarginProps {
+  /** Set whether icon should be recognised by assistive technologies */
+  "aria-hidden"?: boolean;
+  /** Aria label for accessibility purposes */
+  ariaLabel?: string;
+  /** Id passed to the icon. */
+  id?: string;
+  /** The ARIA role to be applied to the Icon */
+  role?: string;
+  /** The message to be displayed within the tooltip */
+  tooltipMessage?: React.ReactNode;
+  /** The position to display the tooltip */
+  tooltipPosition?: TooltipPositions;
+  /** Control whether the tooltip is visible */
+  tooltipVisible?: boolean;
+  /** Override background color of the Tooltip, provide any color from palette or any valid css color value. */
+  tooltipBgColor?: string;
+  /** Override font color of the Tooltip, provide any color from palette or any valid css color value. */
+  tooltipFontColor?: string;
+  /** Overrides the default flip behaviour of the Tooltip */
+  tooltipFlipOverrides?: TooltipPositions[];
+  /** Id passed to the tooltip container, used for accessibility purposes */
+  tooltipId?: string;
+  /**
+   * Icon type
+   *
+   * The full list of types can be seen [here](https://github.com/Sage/carbon/blob/master/src/components/icon/icon-config.js).
```

Comment:
We may want to replace this link or remove it entirely if the generated prop table displays all the possible icon types?

```suggestion
   * The full list of types can be seen [here](https://carbon.sage.com/?path=/docs/icon--list-of-icons#list-of-icons).
```